albatross.ml

```diff
@@ -565,37 +633,37 @@ module Make (S : Tcpip.Stack.V4V6) = struct
             m "init: key_cert failed for %s: %s"
               (Configuration.name_to_str configuration.name)
               s);
-        let state =
-          {
-            error =
-              Some
-                (Fmt.str "Key certificate initialization failed with error: %s"
-                   s);
-            policies = Vmm_trie.empty;
-            configuration;
-          }
+        let error =
+          Status.make `Configuration
+            (Fmt.str "Initialisation error: key_cert failed for %s with %s"
+               (Configuration.name_to_str configuration.name)
+               s)
         in
-        Lwt.return state
+        Lwt.return (dummy_state ~status:(Offline [ error ]) configuration)
     | Ok (key, cert) -> (
         let certificates = `Single ([ cert ], key) in
-        raw_query stack configuration certificates cmd reply >>= function
+        raw_query stack (dummy_state configuration) certificates cmd reply
+        >>= function
         | Ok (_hdr, `Success (`Policies ps)) ->
             let policies =
               List.fold_left
                 (fun trie (name, p) -> fst (Vmm_trie.insert name p trie))
                 Vmm_trie.empty ps
             in
-            let state = { error = None; policies; configuration } in
+            let state = { status = Online; policies; configuration } in
             Lwt.return state
         | Ok w ->
-            Logs.warn (fun m ->
-                m "init: unexpected policy reply for %s: %a"
-                  (Configuration.name_to_str configuration.name)
-                  (Vmm_commands.pp_wire ~verbose:false)
-                  w);
+            let err =
+              Fmt.str "Unexpected policy reply for %s: %a"
+                (Configuration.name_to_str configuration.name)
+                (Vmm_commands.pp_wire ~verbose:false)
+                w
+            in
+            Logs.warn (fun m -> m "%s" err);
+            let error = Status.make `Internal err in
             let state =
```

Comment:
here you'd `state.status <- Status.update Online error;` (instead of the `let state = ...`)